pennylane/compiler/python_compiler/dialects/quantum.py

```diff
@@ -768,7 +1271,45 @@ class QubitUnitaryOp(IRDLOperation):
 
 @irdl_op_definition
 class SampleOp(IRDLOperation):
-    """Sample eigenvalues from the given observable for the current state"""
+    """Sample eigenvalues from the given observable for the current state.
+
+    The ``quantum.sample`` operation represents the measurement process of sampling eigenvalues
+    from an observable on the current quantum state.
+
+    Args:
+        obs (SSAValue): an observable that must be defined by an operation in the local scope.
+
+    Results:
+        samples (SSAValue): The number of samples to draw is determined by the device shots argument in the device initialization operation in the local scope.
+
+
+    .. note::
+
+        The return value type depends on the type of observable provided. Computational
+        basis samples are returned as a 2D array of shape (shot number, number of qubits), with all
+        other obversables the output is a 1D array of lenth equal to the shot number.
+
+    .. note::
+
+        The ``in_data`` field is needed only after bufferization. It is an implementation detail that
+        transform writers are unlikely to be worried about.
+
+    **Example**
+
+    .. code-block:: mlir
+
+        func.func @foo(%q0: !quantum.bit, %q1: !quantum.bit, %shots: i64)
+        {
+            quantum.device shots(%shots) ["rtd_lightning.so", "lightning.qubit", "{my_attr: my_attr_value}"]
+            %obs1 = quantum.compbasis %q0, %q1 : !quantum.obs
+            %samples = quantum.samples %obs1 : tensor<?xf64>
+
+            %obs2 = quantum.pauli %q0[3], %q1[1] : !quantum.obs
+            %samples2 = quantum.samples %obs2 : tensor<?x2xf64>
```

Comment:
```suggestion
            %samples2 = quantum.samples %obs2 : tensor<?xf64>
```
I guess it should be the same as above?